### PR TITLE
Organise structure of problems docs

### LINKED ISSUE #1003: Download links for datasets
The datasets have been modified but do not have download links

New links needed:
- [ ] New SASView
- [ ] Data Assimilation
- [ ] CrystalField
- [ ] GO

Descriptions needed for:
- [ ] CrystalField
- [ ] New SasView

### PR DESCRIPTION
**Pull request by AndrewLister-STFC**
#### Description of Work
Ensure that all problem sets have a descriptions with the following:
 - Download links
 - Information on source of data
 - Information on size of problem -- unknowns and data points.

Currently there is a folder on powell with new versions of everything in generated by:
```
for d in $(ls examples/benchmark_problems); do zip -r ${d}.zip $d; done
for d in $(ls examples/benchmark_problems); do tar -czvf ${d}.tar.gz $d; done
```

Fixes #1003 

#### Testing Instructions

1. Look at the files on powell (called `fitbenchmarking-new` in samelocation as live files)
2. Proof read docs
3. Check links appear to work -- note, this may need to be done after changing the folder name on powell

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: docs/source/users/BenchmarkProblems.rst [hunk 1]
```diff
@@ -146,10 +187,41 @@ The WISH problems find 5 unknown parameters, and fit to 512 data points.
...
+
+These problems (also found in the folder `examples/benchmark_problems/Poisson`) contain
+both simulated and real data measuring particle counts. The real data is ISIS
+muon data, and the simulated datasets have been made to represent counts using
+models provided by both Mantid and Bumps.
+
+These problems have between 4 and 6 unknown parameters, and around 350, 800,
+and 2000 data points for simulated bumps, HIFI_160973, and simulated mantid
+respectively.
+
```
Comment (tyronerees) [inline docs/source/users/BenchmarkProblems.rst:218]:
```suggestion
.. warning::   
   |MantidWarning|
```

File: docs/source/users/BenchmarkProblems.rst [hunk 2]
```diff
@@ -46,10 +48,58 @@ the available fitting methods.
...
+**Download** :download:`.zip <http://numerical.rl.ac.uk/fitbenchmarking/CrystalField.zip>`
+or :download:`.tar.gz <http://numerical.rl.ac.uk/fitbenchmarking/CrystalField.tar.gz>`
+
+This folder (also found in `examples/benchmark_problems/CrystalField`) contains
+a test set for inelastic neutron scattering measurements of transitions between
+crystal field energy levels.
+
+This problem has 8 parameters, and fits around 200 data points. 
+
```
Comment (tyronerees) [inline docs/source/users/BenchmarkProblems.rst:63]:
```suggestion

.. warning::   
   |MantidWarning|
```

**Review by tyronerees — CHANGES_REQUESTED** [review]
I think there's a couple of mantid warnings missing -- otherwise all looks good to me!

File: docs/source/users/BenchmarkProblems.rst (hunk 1 above)
Comment (tyronerees) [reply docs/source/users/BenchmarkProblems.rst]: Sorry, I think I got the whitespace wrong here -- it comes out like: 
image

**Comment by tyronerees** [comment]
I've just noticed that the spacing at the start of the file is a bit wonky too:
image

I can't see how this can be new, but I think it was correct on my first pass through(?)

File: docs/source/users/BenchmarkProblems.rst [hunk 3]
```diff
@@ -146,10 +189,42 @@ The WISH problems find 5 unknown parameters, and fit to 512 data points.
...
+These problems (also found in the folder `examples/benchmark_problems/Poisson`) contain
+both simulated and real data measuring particle counts. The real data is ISIS
+muon data, and the simulated datasets have been made to represent counts using
+models provided by both Mantid and Bumps.
+
+These problems have between 4 and 6 unknown parameters, and around 350, 800,
+and 2000 data points for simulated bumps, HIFI_160973, and simulated mantid
+respectively.
+.. warning::
```
Comment (AndrewLister-STFC) [inline docs/source/users/BenchmarkProblems.rst:220]:
```suggestion

.. warning::
```

File: docs/source/users/BenchmarkProblems.rst [hunk 4]
```diff
@@ -46,10 +48,60 @@ the available fitting methods.
...
+**Download** :download:`.zip <http://numerical.rl.ac.uk/fitbenchmarking/CrystalField.zip>`
+or :download:`.tar.gz <http://numerical.rl.ac.uk/fitbenchmarking/CrystalField.tar.gz>`
+
+This folder (also found in `examples/benchmark_problems/CrystalField`) contains
+a test set for inelastic neutron scattering measurements of transitions between
+crystal field energy levels.
+
+This problem has 8 parameters, and fits around 200 data points. 
+
+.. warning::
```
Comment (AndrewLister-STFC) [inline docs/source/users/BenchmarkProblems.rst:64]:
```suggestion
.. warning::
```

**Review by tyronerees — APPROVED** [review]

**Comment by tyronerees** [comment]
All looks good -- @AndrewLister-STFC, can you amend the directory on Powell?

**Pull request merged**